modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/TableModifyConverterRule.java

```diff
@@ -71,8 +76,27 @@ protected PhysicalNode convert(RelOptPlanner planner, RelMetadataQuery mq, Logic
         IgniteTable igniteTable = relTable.unwrap(IgniteTable.class);
         assert igniteTable != null;
 
+        IgniteDistribution distribution = igniteTable.distribution();
+
+        if (rel.getOperation() == Operation.DELETE) {
+            // To perform the delete, we need a row with key fields only.
+            // Input distribution contains the indexes of the key columns according to the schema (i.e. for the full row).
+            // Here we adjusting distribution keys so that a row containing only the key fields can be read.
+            BitSet keyFields = new BitSet();
+
+            for (int i = 0; i < igniteTable.descriptor().columnsCount(); i++) {
+                ColumnDescriptor column = igniteTable.descriptor().columnDescriptor(i);
+
+                if (column.key()) {
+                    keyFields.set(column.logicalIndex());
+                }
+            }
+
+            distribution = distribution.apply(PlanUtils.sortedValuesIndexMapping(keyFields));
```

Comment:
the same, let's use `org.apache.ignite.internal.sql.engine.util.Commons#trimmingMapping`